slothy/targets/arm_v7m/arch_v7m.py

```diff
@@ -1417,53 +1443,53 @@ def make(cls, src):
         return obj
 
 class ldm_interval(Armv7mLoadInstruction): # pylint: disable=missing-docstring,invalid-name
-    pattern = "ldm<width> <Ra>,<range>"
+    pattern = "ldm<width> <Ra>,<reg_list>"
     inputs = ["Ra"]
     outputs = []
 
     def write(self):
-        reg_from = self.args_out[0]
-        reg_to = self.args_out[-1]
-        self.range = f"{{{reg_from}-{reg_to}}}"
+        regs = ",".join(self.args_out)
+        self.reg_list = f"{{{regs}}}"
         return super().write()
 
 
     @classmethod
     def make(cls, src):
         obj = Armv7mLoadInstruction.build(cls, src)
-        reg_type = Armv7mInstruction._infer_register_type(obj.range_type)
-        num_regs = len(RegisterType.list_registers(reg_type))
-        obj.increment = (obj.range_end-obj.range_start+1) * 4 # word sized loads
-        obj.args_out =  [f"{obj.range_type}{i}" for i in range(obj.range_start, obj.range_end+1)]
+        reg_list_type, reg_list = Armv7mInstruction._expand_reg_list(obj.reg_list)
+
+        obj.args_out = reg_list
         obj.num_out = len(obj.args_out)
         obj.arg_types_out = [RegisterType.GPR] * obj.num_out
-        obj.args_out_restrictions    = [[ f"r{i+j}"  for j in range(0, num_regs-obj.num_out)] for i in range(0, obj.num_out) ]
-        obj.args_out_combinations = [ ( list(range(0, obj.num_out)), [ [ f"r{i+j}" for i in range(0, obj.num_out)] for j in range(0, num_regs-obj.num_out) ] )]
+        available_regs = RegisterType.list_registers(RegisterType.GPR)
+        obj.args_out_combinations =  [ (list(range(0, obj.num_out)), [list(a) for a in itertools.combinations(available_regs, obj.num_out)])]
+        obj.args_out_restrictions = [ None for _ in range(obj.num_out)    ]
```

Comment:
Not precisely on your PR but: I think it would generally be helpful to have some documentation what `args_out_combinations` and `args_out_restrictions` actually are and how they're supposed to be set. On first sight, they can be quite confusing.